projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,313 @@
+                if (tripStops == null) break;
+
+                if (tripStops.getTripId().equals(trip.getTripId()) // stops beside my stop
+                    && tripStops.getStopId() != stopT.getStopId()) {
+
+                  // ArrivalTime HH:mm:ss to double seconds
+                  String[] hhMMss = (tripStops.getArrivalTime().split(":"));
+                  Double arrivalT =
+                      Double.parseDouble(hhMMss[0]) * 3600
+                          + // H
+                          Double.parseDouble(hhMMss[1]) * 60 // m
+                          + Double.parseDouble(hhMMss[2]); // s
+
+                  stopsAround.add(new Stops_around(tripStops, arrivalT));
+                }
+              }
+            }
+          }
+        }
+      }
+
+    } catch (IOException e) {
+      e.printStackTrace();
+    } finally {
+      try {
+        StopTimeInput.close();
+        StopTimeInput2.close();
+        TripsInput.close();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public void Draw_map() {
+
+    // color1 to color2
+    Color c1 = new Color(255, 208, 235);
+    Color c2 = new Color(126, 0, 67);
+
+    // normalize data
+    int maxData = 0;
+    int minData = heatMap[10][10];
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (maxData < heatMap[i][j]) maxData = heatMap[i][j];
+        if (heatMap[i][j] != 0 && minData > heatMap[i][j]) minData = heatMap[i][j];
+      }
+    }
+    System.out.println("max= " + maxData + "min" + minData);
+
+    // create image
+    /** ToDo: try different size of image */
+    final BufferedImage image =
+        new BufferedImage(
+            heatMap[0].length - 1000, heatMap.length - 1000, BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics2D = image.createGraphics();
+
+    // draw the map
+
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (heatMap[i][j] == 0) System.out.println("heatmap=0 error");
+
+        float value = (float) (heatMap[i][j] - minData) / (float) (maxData - minData);
+
+        int r = (int) (((float) c2.getRed() - (float) c1.getRed()) * value + (float) c1.getRed());
+        int g =
+            (int) (((float) c2.getGreen() - (float) c1.getGreen()) * value + (float) c1.getGreen());
+        int b =
+            (int) (((float) c2.getBlue() - (float) c1.getBlue()) * value + (float) c1.getBlue());
+
+        Color c = new Color(r, g, b);
+        graphics2D.setPaint(c);
+        graphics2D.fillRect(i, j, 2, 2);
+      }
+    }
+
+    graphics2D.dispose();
+    // save file
+    boolean img = false;
+    try {
+      img = ImageIO.write(image, "png", new File(imagePath.get()));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    if (img == false) System.out.println("can't create img");
```

Comment:
Type of `img` variable is boolean.  You can use: 
```
if (!img) {
  // do something
}
```
P.s. `img` isn't a good name for the boolean variable.